### ISSUE TITLE
MWE provokes "Missing character" warning and fails validation (PDF/A-4f)

### ISSUE THREAD

**Issue by t-b** [TeXhackse/LaTeX-ZUGFeRD#20]
Related to https://github.com/TeXhackse/LaTeX-ZUGFeRD/issues/15.

I'm trying to come up with MWE which passes XML+PDF validation.

```tex
\DocumentMetadata{
  pdfstandard=A-4F,
  pdfversion=2.0,
  lang=de,
}

\documentclass{scrartcl}

\usepackage[format=xrechnung3.0]{zugferd}

\SetZUGFeRDData{%
	id=1337,
	unit=hour,
	tax/category=S,
	seller/name = {Seller name},
	seller/postcode = {12345},
	seller/city ={MeineStadt},
	seller/country = DE,
	seller/address = {Address 1},
	seller/vatid = {DE123456789},
	seller/contact= {Me\\+4900000000\\a@b.de},
	seller/email = {a@bb.de},
	buyer/reference = {buyer-reference},
	buyer/name = {Buyer Name},
	buyer/postcode = {12345},
	buyer/city ={DeineStadt},
	buyer/country = DE,
	buyer/vatid = {DE123456789},
	buyer/email = {invoice@example.org},
	due-date={20240118},
	payment-means / type = 1,
	delivery-date=auto
}

\ExplSyntaxOn
\newcommand{\foo}
{
\begin{ZUGFeRD}
\zugferd_write_Item:ennnnnn
  {}       % additional local options like tax/rate=19,tax/category=S
  {1}      % LineID
  {}       % ItemID
  {desc}   % ItemName
  {31.98}  % NetPriceProductTradePrice
  {2}      % BilledQuantity
  {63.78}  % LineTotalAmount

\zugferd_startInvoiceSums:

\zugferd_write_TaxEntry:nnnn
  {S}     % tax category code
  {19}    % tax rate in %
  {63.78} % basis amount the tax applies to
  {12.12} % tax amount

\zugferd_write_Summation:nnnnnnnn
{63.78}   %  LineTotalAmount
{0}       %  ChargeTotalAmount
{0}       %  AllowanceTotalAmount
{63.78}   %  TaxBasisTotalAmount
{12.12}   %  TaxTotalAmount
{75.90}   %  GrandTotalAmount
{0}       %  TotalPrepaidAmount
{75.90}   %  DuePayableAmount

\zugferd_stopInvoiceSums:
\end{ZUGFeRD}
}
\ExplSyntaxOff

\begin{document}
Output
\foo{}
\end{document}
```

but that gives the lualatex warning

```
Missing character: There is no 
 (U+000A) in font [lmroman10-regular]:+tlig;!
] (./test.aux)
```

and mustang complains with

```
  <pdf>ValidationResult [flavour=4f, totalAssertions=189, assertions=[TestAssertion [ruleId=RuleId [specification=ISO 19005-4:2020, clause=6.2.10.9, testNumber=1], status=failed, message=A PDF/A-4 compliant document shall not contain a reference to the .notdef glyph from any of the text showing operators, regardless of text rendering mode, in any content stream, location=Location [level=CosDocument, context=root/document[0]/pages[0](12 0 obj PDPage)/contentStream[0](13 0 obj PDContentStream)/operators[8]/usedGlyphs[6](KNLTRO+LMRoman10-Regular KNLTRO+LMRoman10-Regular 0 0 0 false)], locationContext=null, errorMessage=The document contains a reference to the .notdef glyph]], isCompliant=false]
```

Maybe the package could also ship a MWE passing validation? Yes I now that `DEMO-rechnung-zugferd.tex` is being shipped (thanks for that!) and passes validation. But that is not really minimal.

**Comment by TeXhackse** [comment]
I did not yet provide a test case for PDF 2.0, but you are right. I should add that one. Anyway this rather is an extra `\newline` and I was planning to increase the testcases there anyway. This is rather connected to #8 than 15.

**Comment by t-b** [comment]
Thanks for the correction, I've not seen #8.

**Comment by t-b** [comment]
Just to add another pointer:

This seems to be irrelevant of the pdfstandard and pdfversion. I get the same issue with

```
@@ -1,6 +1,6 @@
 \DocumentMetadata{
-  pdfstandard=A-4F,
-  pdfversion=2.0,
+  pdfstandard=A-3B,
+  pdfversion=1.5,
   colorprofiles = {A = FOGRA39L_coated.icc, X = FOGRA39L_coated.icc},
   lang=de,
 }
```

**Comment by t-b** [comment]
> I'm not sure that make sense to use pdf 2.0 + A-4f with zugferd. The standard clearly speaks everywhere about pdf/A-3.

As mentioned in https://github.com/TeXhackse/LaTeX-ZUGFeRD/issues/20#issuecomment-2470446710 I can reproduce the bug also with PDF 1.5 and PDF A-3B.

I'm happy to adapt pdfversion and pdfstandard to whatever fits best.

**Comment by TeXhackse** [comment]
@u-fischer thanks. I only used a-3b anyway. I added a test file matching a-4f just in case there might be something failing and because I am interested if the validators will continue to accept it, as they do at the moment. 

Anyway as mentioned here before It was a leftover char which was already removed a week ago, but I now published the release containing this + added test addressing this issue so this should not happen again.
